client/java/src/main/java/io/openlineage/client/OverwriteConfig.java

```diff
@@ -0,0 +1,75 @@
+/*
+/* Copyright 2018-2024 contributors to the OpenLineage project
+/* SPDX-License-Identifier: Apache-2.0
+*/
+
+package io.openlineage.client;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Contains methods which allows overwriting values of a config object with values from another
+ * object
+ */
+public interface OverwriteConfig<T> {
```

Comment:
I see you've added this mechanism, but what's the actual purpose of it? Can't we just create new config instance rather than making config mutable if we need to merge config entries from multiple sources?

Reply:
I think we can go with the approach you describe: change the interface to return a new value instead
will prepare a change for this.